### ISSUE TITLE
How does klar work?

### ISSUE THREAD

**Issue by stacygohyunsi** [optiopay/klar#156]
Does klar pull and download the image locally before calling an API to clair to scan it?

**Comment by adamcohen** [comment]
No, it doesn't pull the image locally, it only [pulls the manifest file](https://github.com/optiopay/klar/blob/v2.4.0/docker/docker.go#L315-L341). It then reads the layer data from the manifest file, [loops over each layer](https://github.com/optiopay/klar/blob/v2.4.0/clair/api.go#L69) in the image, and sends a [POST](https://github.com/optiopay/klar/blob/v2.4.0/clair/api.go#L85) request to the `clair` API endpoint with a `Path` component which contains a URL to the layer data in the container registry. Clair then reads the layer data directly from the registry.  This means that `klar` doesn't have to actually push the data to `clair`, it only passes a references to the data which `clair` is responsible for pulling.
